packages/app/src/background/services/injector/Injector.ts

```diff
@@ -36,6 +36,26 @@ export class InjectorService {
     return InjectorService.INSTANCE;
   }
 
+  isApproved = (payload: unknown, { urlOrigin }: IZkMetadata): unknown => {
+    const { isApproved } = this.injectorHandler.getConnectionApprovalData({ urlOrigin });
+
+    if (!isApproved) {
+      throw new Error("CryptKeeper: Origin is not approved");
+    }
+
+    return payload;
+  };
+
+  isConnected = async (payload: unknown, { urlOrigin }: IZkMetadata): Promise<unknown> => {
+    const connectedIdentity = await this.injectorHandler.getConnectedIdentityMetadata({}, { urlOrigin });
+
+    if (connectedIdentity.urlOrigin !== urlOrigin) {
```

Comment:
When this is happening? because if there is not connected identity at all the error would be failed from the  `getConnectedIdentityMetadata` which is`CryptKeeper: identity metadata is not found` and if there is a connected identity came out of the `getConnectedIdentityMetadata` it would be always equal to `urlOrigin`.

Reply:
Yes, it can be removed